hr_timesheet_begin_end/models/account_analytic_line.py

```diff
@@ -15,16 +16,22 @@ class AccountAnalyticLine(models.Model):
     time_start = fields.Float(string="Begin Hour")
     time_stop = fields.Float(string="End Hour")
 
-    @api.constrains("time_start", "time_stop", "unit_amount")
-    def _check_time_start_stop(self):
+    # Override to be a computed field.
+    unit_amount = fields.Float(
+        compute="_compute_unit_amount",
+        store=True,
+        readonly=False,
+    )
```

Comment:
This reproduces as closely as possible the old behaviour. Technically `readonly=False` isn't really desirable here, I think, but it matches how it used to be when this used onchange (and there is a constraint to make sure the two stay in sync, anyway).

Reply:
as this model is also used for other things than timesheets, it is very important that `unit_amount` must not be computed if `time_start` and `time_stop` are both not set. since it is `store=True` and `readonly=False`, it should work as expected. adding a test for this would be good.

Reply:
Did this in #692 in the `create` method. Was a bit tricky to get right.